include/mbgl/util/geo.hpp

```diff
@@ -103,14 +107,18 @@ class LatLngBounds {
 
     // Return a bounds that may serve as the identity element for the extend operation.
     static LatLngBounds empty() {
-        LatLngBounds bounds = world();
+        LatLngBounds bounds = LatLngBounds::world();
         std::swap(bounds.sw, bounds.ne);
         return bounds;
     }
 
     // Constructs a LatLngBounds object with the tile's exact boundaries.
     LatLngBounds(const CanonicalTileID&);
 
+    explicit operator bool() const {
+        return (sw.latitude <= ne.latitude) && (sw.longitude <= ne.longitude);
```

Comment:
My understanding of [LatLngBounds::extend](https://github.com/mapbox/mapbox-gl-native/blob/master/include/mbgl/util/geo.hpp#L129) is that it preserves the assertion above. Also, this should be fine as long as the coordinates used are unwrapped.

Nonetheless, we can add a unit test to verify 👍